### PR TITLE
LRCI-1980 Add 'Build.getBuildResultsJSONObject' methods to build objects

### PR DESCRIPTION
**Pull request by kenjiheigel**
https://issues.liferay.com/browse/LRCI-1980

### CODE REVIEW

File: modules/test/jenkins-results-parser/src/main/java/com/liferay/jenkins/results/parser/BaseBuild.java [hunk 1]
```diff
@@ -354,6 +355,91 @@ public int getBuildNumber() {
...
+		}
+		else {
+			for (String testStatus : testStatuses) {
+				try {
+					testResults.addAll(getTestResults(testStatus));
+				}
+				catch (RuntimeException runtimeException) {
+					System.out.println(runtimeException.getMessage());
+				}
+			}
```
Comment (pyoo47) [inline modules/test/jenkins-results-parser/src/main/java/com/liferay/jenkins/results/parser/BaseBuild.java:387]: @kenjiheigel This looks a little off to me. What exception are you anticipating here?

Comment (kenjiheigel) [reply modules/test/jenkins-results-parser/src/main/java/com/liferay/jenkins/results/parser/BaseBuild.java:387]: @pyoo47, there's some cases where the Axis builds fail unexpectedly (ClosedChannelException) and doesn't generate the `*/testReport/api/json` page, which can result in a RuntimeException. 

This was a fix that I had to add into the beanshell code in [LRCI-1950](https://issues.liferay.com/browse/LRCI-1950) 
https://github.com/liferay/liferay-jenkins-ee/commit/4b6e02a5b18cabaad30e3bf05e9040d6871dfa17#diff-c393e5a7bf3f48c54b462439c24090ca4947b3ea9acc3c50c08c501dccbf41ba

**Comment by pyoo47** [comment]
ci:forward

File: modules/test/jenkins-results-parser/src/main/java/com/liferay/jenkins/results/parser/BaseBuild.java (hunk 1 above)
Comment (pyoo47) [reply modules/test/jenkins-results-parser/src/main/java/com/liferay/jenkins/results/parser/BaseBuild.java:387]: I think that in a situation like this you should identify the source of the RuntimeException and catch it at some lower level or change how it's handled altogether. Let's discuss this further in person and I can explain exactly what I mean. For now, I'll push this forward but I think this should be revisited and handled properly.

**Pull request closed without merging**